gene2phenotype_project/gene2phenotype_app/serializers/user.py

```diff
@@ -176,21 +447,151 @@ class AuthSerializer(serializers.Serializer):
     password = serializers.CharField(
         style={'input_type': 'password'},
-        trim_whitespace=False
-    )    
+        trim_whitespace=False,
+        write_only=True
+    )
+    tokens = serializers.SerializerMethodField()
+    
+    def get_tokens(self, obj):
+        """
+            Returns the authentication tokens for a given user.
+
+            This method retrieves the `refresh` and `access` tokens for the user based on their email.
+
+            Parameters:
+                - obj (dict): A dictionary containing user information, including the user's email.
+
+            Returns:
+                - dict: A dictionary containing the `refresh` and `access` tokens associated with the 
+                user, typically used for authentication purposes.
+
+            Raises:
+                - User.DoesNotExist: If no user with the provided email exists in the database.
+        """
+
+        user = User.objects.get(email=obj['email'])
+
+        if user:
+            return {
+                'refresh': user.tokens()['refresh'],
+                'access': user.tokens()['access']
+            }
+    
     def validate(self, attrs):
+        """
+            Validates the user credentials and checks account status.
+
+            This method authenticates the user by verifying the provided username and password. It also
+            checks whether the user's account is disabled. If successful, it returns the user's email, 
+            username, and authentication tokens.
+
+            Parameters:
+                - attrs (dict): A dictionary containing user input data, specifically 'username' and 
+                'password'.
+
+            Returns:
+                - dict: A dictionary containing the user's email, username, and tokens if authentication 
+                is successful.
+
+            Raises:
+                - AuthenticationFailed: If the authentication fails due to incorrect username or password, 
+                or if the user's account has been disabled.
+
+            Note:
+                The user must not be deleted (i.e., `user.is_deleted` must be `False`). If the account is 
+                deleted, an error message is raised with instructions to contact support.
+        """
+
         username = attrs.get('username')
         password = attrs.get('password')
+
+        if username and password:
         
-        user = authenticate(
-            request=self.context.get('request'),
-            username=username,
-            password=password
-        )
+            user = authenticate(
+                request=self.context.get('request'),
+                username=username,
+                password=password
+            )
+            if not user:
+                raise AuthenticationFailed("Username or password is incorrect")
+
+            if user.is_deleted:
+                raise AuthenticationFailed('Account disabled. Please contact Admin at g2p-help@ebi.ac.uk')
+
```

Comment:
When a user logs in the column 'last_login' used to be updated automatically by `django.contrib.auth`
Now this column is not being updated. Can you add the following code here to make sure the column keeps being updated?
```
if request_content:
    login(request_content, user)
```
And add `login` to line 6.

Reply:
This has been updated without creating a session. Let me know if there are any other issues.